codes/codes_test.go

```diff
@@ -91,3 +91,12 @@ func (s) TestUnmarshalJSON_MarshalUnmarshal(t *testing.T) {
 		}
 	}
 }
+
+func (s) TestUnmarshalJSON_InvalidIntegerCode(t *testing.T) {
+	var got Code
+	wantErr := "invalid code: 200" // for integer invalid code, expect integer value in error message
```

Comment:
Nit: `const`

Reply:
Or `wantErr := errors.New("invalid code: 200")`. If you do this, then the error comparison at the bottom can be a simple `err != wantErr`

Reply:
Thanks. Changed as suggested. However, looks error object equality check doesn't so I am comparing the string. I also tried `cmp.Equal` and `errors.Is` but both fail

Reply:
Why does direct error value comparison or `errors.Is` not work?

Reply:
I did some research and this is what I found. errors.New returns a pointer. Go pointers never compare equal unless they point to the same object. That is, unless they’re literally the same memory address.

And that won’t be the case here, because they point to two distinct instances of the error struct even though they contain the same error message.

I think in our case we basically just want to test if the error string contains the same invalid integer code so may be comparing just error message is acceptable even though. The test with this approach, however is fragile if someone change the error string in  future. Alternatively, we can define the named error value with a generic message like `input integer code value is greater than allowed code values`

Let me know what you think

Reply:
We could go with the following approach which is also quite common in our repo:

```golang
	const wantErr := "invalid code: 200"

	var got Code
	if err := got.UnmarshalJSON([]byte("200")); !strings.Contains(err.Error(), wantErr) {
		t.Errorf("got.UnmarshalJSON(200) = %v; wantErr: %v", err, wantErr)
	}
```

Reply:
Done